cli/internal/client/client.go

```diff
@@ -30,21 +32,55 @@ func (api *ApiClient) SetToken(token string) {
 
 // New creates a new ApiClient
 func NewClient(baseUrl string, logger hclog.Logger, turboVersion string) *ApiClient {
-	return &ApiClient{
-		baseUrl:      baseUrl,
-		turboVersion: turboVersion,
+	client := &ApiClient{
+		baseUrl:          baseUrl,
+		turboVersion:     turboVersion,
+		CurrentFailCount: 0,
 		HttpClient: &retryablehttp.Client{
 			HTTPClient: &http.Client{
-				Timeout: time.Duration(60 * time.Second),
+				Timeout: time.Duration(20 * time.Second),
 			},
-			RetryWaitMin: 10 * time.Second,
-			RetryWaitMax: 20 * time.Second,
-			RetryMax:     5,
-			CheckRetry:   retryablehttp.DefaultRetryPolicy,
+			RetryWaitMin: 2 * time.Second,
+			RetryWaitMax: 10 * time.Second,
+			RetryMax:     2,
 			Backoff:      retryablehttp.DefaultBackoff,
 			Logger:       logger,
 		},
 	}
+	return client
+}
+
+func (client *ApiClient) retryCachePolicy(resp *http.Response, err error) (bool, error) {
+	if err != nil {
+		if v, ok := err.(*url.Error); ok {
```

Comment:
It might be clearer in these situations to use something like [`errors.Is`](https://pkg.go.dev/errors#Is) or [`errors.As`](https://pkg.go.dev/errors#As). Some examples [here](https://github.com/uber-go/guide/blob/master/style.md#error-types)